libcnb-cargo/src/main.rs

```diff
@@ -0,0 +1,190 @@
+use std::path::PathBuf;
+
+fn main() {
+    setup_logging();
+
+    match cli::setup_cli_parsing().get_matches().subcommand() {
+        ("libcnb", Some(matches)) => match matches.subcommand() {
+            ("package", Some(matches)) => handle_libcnb_package(matches),
+            // This should never be reached since clap will catch unknown subcommands for us
+            _ => unimplemented!("Only the \"package\" subcommand is currently implemented!"),
+        },
+        // This should never be reached since clap will catch unknown subcommands for us
+        _ => unimplemented!("Only the \"libcnb\" subcommand is currently implemented!"),
+    }
+}
+
+#[allow(clippy::too_many_lines)]
+fn handle_libcnb_package(matches: &ArgMatches) {
+    let cargo_profile = if matches.is_present("release") {
+        CargoProfile::Release
+    } else {
+        CargoProfile::Dev
+    };
+
+    let target_triple = match matches.value_of("target") {
+        None => {
+            error!("Could not determine target triple!");
+            std::process::exit(1);
+        }
+        Some(target_triple) => target_triple,
+    };
+
+    let current_dir = match std::env::current_dir() {
+        Ok(current_dir) => current_dir,
+        Err(io_error) => {
+            error!("Could not determine current directory: {}", io_error);
+            std::process::exit(1);
+        }
+    };
+
+    info!("Reading buildpack metadata...");
+    let buildpack_data = match read_buildpack_data(&current_dir) {
+        Ok(buildpack_data) => buildpack_data,
+        Err(error) => {
+            match error {
+                BuildpackDataError::IoError(io_error) => {
+                    error!("Unable to read buildpack metadata: {}", io_error);
+                    error!("Hint: Verify that a readable file named \"buildpack.toml\" exists at the root of your project.");
+                }
+                BuildpackDataError::DeserializationError(deserialization_error) => {
+                    error!(
+                        "Unable to deserialize buildpack metadata: {}",
+                        deserialization_error
+                    );
+                    error!("Hint: Verify that your \"buildpack.toml\" is valid.");
+                }
+            }
+
+            std::process::exit(1);
+        }
+    };
+
+    info!(
+        "Found buildpack {} with version {}.",
+        buildpack_data.buildpack_toml.buildpack.id, buildpack_data.buildpack_toml.buildpack.version
+    );
+
+    let cargo_metadata = match MetadataCommand::new()
+        .manifest_path(&current_dir.join("Cargo.toml"))
+        .exec()
+    {
+        Ok(cargo_metadata) => cargo_metadata,
+        Err(error) => {
+            error!("Could obtain metadata from Cargo: {}", error);
+            std::process::exit(1);
+        }
+    };
+
+    let output_path = matches.value_of("output-path").map_or_else(
+        || {
+            cargo_metadata
+                .target_directory
```

Comment:
The default directory location generated here is of form:
`target/debug/libcnb-examples_basics_0.1.0/`

Whereas the assets for the cross-compiled builds are inside the appropriate target triple directory:
`target/x86_64-unknown-linux-musl/debug/`

So if we're being consistent, the path should probably be something like:
`target/x86_64-unknown-linux-musl/debug/libcnb-examples_basics_0.1.0/`

Although that path is now very long, and annoying to deal with if people are calling pack by hand on the CLI.

What do you think about removing both the {debug/release} and buildpack version components from the directory path?

IMO including them:
- doesn't really add much value (packaging is super fast, overwriting isn't an issue for switching back and forth between debug and release; we don't really need the version in the path etc)
- makes a bunch of use-cases more annoying (eg manually calling pack or other non-Rust-library usages, and having to adjust paths repeatedly, as versions change etc)
- means over time lots of old version directories will accumulate, affecting things like tab-complete for non-Rust-library usages

Removing those would give something like:
```
target/libcnb-examples_basics/
target/libcnb-examples_ruby/
```

Or if we wanted to not clutter up target, then something like:

```
target/dist/libcnb-examples_basics/
target/dist/libcnb-examples_ruby/
```

Reply:
I think both version and Cargo profile are important. I can see removing the version from the path since it can be read from the `buildpack.toml`, but the Cargo profile used to compile the buildpack should be there IMO.


WDYT about smth. like this?
```
target/buildpack/debug/libcnb-examples_basics
target/buildpack/debug/libcnb-examples_ruby
```

Reply:
> It's still a long path, but it will be stable across subsequent invocations

Yeah this was my primary concern previously. The proposed path works for me :-)

Reply:
Added here: 53265524c3bcd4bea4a56fa9ad6efceb11cb7a72